velox/exec/RowContainer.cpp

```diff
@@ -385,6 +389,19 @@ int32_t RowContainer::findRows(folly::Range<char**> rows, char** result) {
   return numRows;
 }
 
+void RowContainer::appendNextRow(char* current, char* nextRow) {
+  NextRowVector*& nextRowArrayPtr = getNextRowVector(current);
+  if (!nextRowArrayPtr) {
+    nextRowArrayPtr =
+        new (stringAllocator_->allocateFromPool(kNextRowVectorSize))
```

Comment:
Do not use `allocateFromPool` as `kNextRowVectorSize` is small, use the normal `allocate` and take `Header::begin`.  When release you can use `HashStringAllocator::headerOf` to get the `Header*`